velox/exec/RowContainer.h

```diff
@@ -161,6 +161,7 @@ class RowContainer {
  public:
   static constexpr uint64_t kUnlimited = std::numeric_limits<uint64_t>::max();
   using Eraser = std::function<void(folly::Range<char**> rows)>;
+  friend class PrefixSort;
```

Comment:
Can we avoid using 'friend'?

Reply:
This is what I have been struggling with for a long time. The reason for using friend is: PrefixSort needs to get data from RowContainer, but I don't want RowContainer to be dependent on the code of PrefixSort. I understand that the current solution that can take into account both performance and code dependencies can only use friend. Do you have any better suggestions?

@mbasmanova

Reply:
> PrefixSort needs to get data from RowContainer, but I don't want RowContainer to be dependent on the code of PrefixSort

@skadilover I haven't read the code yet, but maybe you can explain a bit what data you need and why can't we have some sort of extractXxx API in RowContainer for the purpose.

Reply:
@mbasmanova  The key point is that: when extracting data, we need to rely on PrefixSort logic (normalize/variable length  process ..) , this part of logic will be a little complex.